### ISSUE TITLE
variables with value?

### ISSUE THREAD

**Issue by mihai-satmarean** [willthames/ansible-inventory-grapher#37]
Hi, 
the graphing tool is nice, 
I am trying to get the values of variables too, is it that something easy to achieve?
Thanks!

**Comment by willthames** [comment]
It's not impossible, just not sure how easy it would be to represent environments with a lot of info in variables. And then there's the question of what to do with the vaulted variables.

I'm unlikely to implement this as I find where the variable is set is usually enough to work out why a variable isn't set to what I expect it to be, which is what I typically use it for! 

If someone does want to implement this, I'd be happy to take a look at including it.

**Comment by mihai-satmarean** [comment]
Thanks!
I thought it would be a matter of just addressing another key value pair

2018-04-10 11:57 GMT+02:00 Will Thames <notifications@github.com>:

**Comment by eking-go** [comment]
Hello.

May be possible print content only one predefined variable (if present in inventory file for this host)?

Fo example, use variable name 'comment' or 'aig'...

**Comment by eking-go** [comment]
Hello. I add this feature in branch variables_with_value in my fork
Diff:
```
42c42
< {% if node.vars and showvars %}<hr/><tr><td><font face="Times New Roman, Bold" point-size="14">{% for var in node.vars|sort %}{{var}}<br/>{%endfor %}</font></td></tr>{% endif %}
---
> {% if node.vars and showvars %}<hr/><tr><td><font face="Times New Roman, Bold" point-size="14">{% for var in node.vars|sort %}{{var}}{% if var in visible_vars %} = {{node.vars[var]}}{% endif %}<br/>{%endfor %}</font></td></tr>{% endif %}
51c51
< {% if node.vars and showvars %}<hr/><tr><td><font face="Times New Roman, Bold" point-size="14">{% for var in node.vars|sort %}{{var}}<br/>{%endfor %}</font></td></tr>{% endif %}
---
> {% if node.vars and showvars %}<hr/><tr><td><font face="Times New Roman, Bold" point-size="14">{% for var in node.vars|sort %}{{var}}{% if var in visible_vars %} = {{node.vars[var]}}{% endif %}<br/>{%endfor %}</font></td></tr>{% endif %}
92a93,94
>     parser.add_option('--visible-vars', default=[], dest='visible_vars', action='append', type='string',
>                       help='Visible variables with values')
143c145,146
<                              showvars=options.showvars)
---
>                              showvars=options.showvars,
>                              visible_vars=options.visible_vars)
```
Now possible to show values of variables if variable listed in --visible-vars. Not for all variables, so, do not print secret values (if it not need)

**Comment by willthames** [comment]
Fixed by #39

**Comment by willthames** [comment]
I've also just added a `--show-all-values` flag so that you can show all values, not just one at a time (but that would need a vault password for secrets, and you might not want to show all the secrets)

**Comment by willthames** [comment]
`--show-all-values` will be available in 2.5.0
